config/routes.rb

```diff
@@ -351,6 +351,7 @@
       collection do
         post 'add_existing_annotation'
         patch 'update_annotation'
+        post 'edit'
```

Comment:
Use of `resources` is a common enough convention that it is not worth restating. It's also not the convention we follow for the rest of the routes, and not something that should be deviated from.

You're having trouble because you didn't actually replicate the route that `resources` generates; instead, you added a new one that looks similar, but is actually different. The correct path helper does exist: it is `edit_annotation_path`. See http://guides.rubyonrails.org/routing.html#path-and-url-helpers for details.